### DISCUSSION TITLE
Combobox: Provide way to pass element ref for anchor

### DISCUSSION THREAD

**Discussion by adarshpastakia** [tailwindlabs/headlessui#3415] [Ideas]
When using the multiple option we wrap the combox input in a flex div which displays the selected values as chips causing the input to shift. However since there is no way of anchoring the combobox to the wrapping div the floating dropdown shifts along with the input which isn't very desirable, for now as work around we are using floating-ui to override the default behaviour.

```
<div className="flex gap-2 p-1 outline">
  <span className="inline-block bg-slate-100">Selected 1</span>
  <ComboxboxInput className="flex-1 min-w-24"/>
</div>
```

It would desirable to have the ability to pass an element ref via the anchor object to override the default floating anchor.

Also we receive many user complaints of having to either close the dropdown by pressing escape and then press enter to submit the form, or click the submit button after selecting a value. An option to closeOnSelect would be helpful, thereby the user can hit enter twice to select and submit.

**Comment by dlindahl** [comment]
Related discussion: https://github.com/tailwindlabs/headlessui/discussions/3347
